packages/nova-types/src/nova-localization.ts

```diff
@@ -0,0 +1,29 @@
+const LocalizedString = Symbol("LocalizedString");
+
+export type StringWithPlaceholders<T extends Record<string, string | number>> =
+  {
+    [LocalizedString]: T;
+  };
+
+export type Placeholders<T> = T extends StringWithPlaceholders<infer P>
+  ? keyof P extends never // Make sure that the placeholder object is not empty
+    ? never
+    : P
+  : never;
+
+type UseFormat = () => (
```

Comment:
Please add a docstring explaining the semantics of the arguments here. Also, let's try to name them better than `value` and `values`. Currently, reader can't saay how these two differ.

Reply:
Also, I think you intended to export `UserFormat` as it is used in an exported type below.